presto-main/src/test/java/com/facebook/presto/sql/planner/TestDomainTranslator.java

```diff
@@ -1155,173 +1163,99 @@ public void testExpressionConstantFolding()
-        testSimpleComparison(lessThanOrEqual(C_INTEGER, bigintLiteral(Integer.MIN_VALUE - 1L)), C_INTEGER, Range.lessThan(INTEGER, (long) Integer.MIN_VALUE));
-
-        // less than
-        testSimpleComparison(lessThan(C_INTEGER, bigintLiteral(-2L)), C_INTEGER, Range.lessThan(INTEGER, -2L));
-        testSimpleComparison(lessThan(C_INTEGER, bigintLiteral(Integer.MIN_VALUE - 1L)), C_INTEGER, Range.lessThan(INTEGER, (long) Integer.MIN_VALUE));
-
-        // equal
-        testSimpleComparison(equal(C_INTEGER, bigintLiteral(Integer.MIN_VALUE - 1L)), C_INTEGER, Domain.none(INTEGER));
-        testSimpleComparison(equal(C_INTEGER, bigintLiteral(-2L)), C_INTEGER, Range.equal(INTEGER, -2L));
-
-        // not equal
-        testSimpleComparison(notEqual(C_INTEGER, bigintLiteral(Integer.MAX_VALUE + 1L)), C_INTEGER, Domain.notNull(INTEGER));
-        testSimpleComparison(notEqual(C_INTEGER, bigintLiteral(2L)), C_INTEGER, Domain.create(ValueSet.ofRanges(Range.lessThan(INTEGER, 2L), Range.greaterThan(INTEGER, 2L)), false));
-
-        // is distinct from
-        testSimpleComparison(isDistinctFrom(C_INTEGER, bigintLiteral(Integer.MAX_VALUE + 1L)), C_INTEGER, Domain.all(INTEGER));
-        testSimpleComparison(isDistinctFrom(C_INTEGER, bigintLiteral(2L)), C_INTEGER, Domain.create(ValueSet.ofRanges(Range.lessThan(INTEGER, 2L), Range.greaterThan(INTEGER, 2L)), true));
-    }
-
-    @Test
-    public void testDecimalComparedToWiderDecimal()
-            throws Exception
-    {
-        // greater than or equal
-        testSimpleComparison(greaterThanOrEqual(C_DECIMAL_4_2, decimalLiteral("44.555678")), C_DECIMAL_4_2, Range.greaterThan(createDecimalType(4, 2), shortDecimal("44.55")));
-        testSimpleComparison(greaterThanOrEqual(C_DECIMAL_4_2, decimalLiteral("99.99")), C_DECIMAL_4_2, Range.greaterThanOrEqual(createDecimalType(4, 2), shortDecimal("99.99")));
-        testSimpleComparison(greaterThanOrEqual(C_DECIMAL_4_2, decimalLiteral("9999.999")), C_DECIMAL_4_2, Range.greaterThan(createDecimalType(4, 2), shortDecimal("99.99")));
+        testSimpleComparison(greaterThanOrEqual(columnSymbol, literalConstructor.apply(literalValues.getPositive())), columnSymbol, Range.greaterThanOrEqual(columnType, columnValues.getPositive()));
+        testSimpleComparison(greaterThanOrEqual(columnSymbol, literalConstructor.apply(literalValues.getNegative())), columnSymbol, Range.greaterThanOrEqual(columnType, columnValues.getNegative()));
+        testSimpleComparison(greaterThanOrEqual(columnSymbol, literalConstructor.apply(literalValues.getMax())), columnSymbol, Range.greaterThan(columnType, columnValues.getMax()));
+        testSimpleComparison(greaterThanOrEqual(columnSymbol, literalConstructor.apply(literalValues.getMin())), columnSymbol, Range.greaterThanOrEqual(columnType, columnValues.getMin()));
+        if (literalValues.isFractional()) {
```

Comment:
is this special fraction case required? Can't
`positive` and `negative` be fractional itself?